### PR TITLE
Allow PHP ^8.0, add Github Actions

### PR DESCRIPTION
**Pull request by trejjam**
Hi,

this PR adds support for PHP 8.0 in composer.json. And it also replaces Travis with Github Actions (based on [Contributte Actions](https://github.com/contributte/bare/blob/master/.github/workflows/main.yaml)).

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
(I'm not the owner of this repo)

Hi, you have a much higher chance of merging a PR if you create two separate ones, each for separate concern this PR includes.

**Comment by trejjam** [comment]
Hi, thank you for the suggestion. I am aware of that. The reason why it is in single PR is the response time of the Travis CI (sometimes takes hours). The scope of `Allow PHP ^8.0` is only in the `composer.json`, and Github Actions already has support for PHP 8.0 (with makes the change in the `composer.json` mandatory).

**Comment by trejjam** [comment]
See Github Actions executed in the fork scope: [trejjam/roave-no-floaters#1](https://github.com/trejjam/roave-no-floaters/pull/1)

**Review by joshuaslaney — APPROVED** [review]

**Review by joshuaslaney — APPROVED** [review]

**Review by joshuaslaney — APPROVED** [review]

**Review by joshuaslaney — APPROVED** [review]
[Slaneyjosh14@gmail.com ]()

**Review by joshuaslaney — APPROVED** [review]
Slaneyjosh14@gmail.com [](slaneyjosh14@gmail.com)

**Comment by trejjam** [comment]
@joshuaslaney, please stop. This is not how things works...

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  composer-install: "composer update --no-interaction --no-progress --no-suggest --prefer-dist --prefer-stable"
+
+jobs:
+  qa:
+    name: "Quality assurance"
+    runs-on: "${{ matrix.operating-system }}"
+
+    strategy:
+      matrix:
+        php-version: ["7.4"]
```
Comment (Ocramius) [inline .github/workflows/main.yml:26]: IMO to be tested only with `8.0`

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      - name: "Coding Standard"
+        run: "make cs"
+
+  static-analysis:
+    name: "Static analysis"
+    runs-on: "${{ matrix.operating-system }}"
+
+    strategy:
+      matrix:
+        php-version: ["7.4"]
```
Comment (Ocramius) [inline .github/workflows/main.yml:85]: IMO to be tested only with `8.0`

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.PHONY: install qa cs phpstan tests mutation-tests
+
+install:
+	composer update
```
Comment (Ocramius) [inline Makefile:4]: Should be `composer install`

File: composer.json [hunk 4]
```diff
@@ -5,16 +5,16 @@
         "MIT"
     ],
     "require": {
-        "php": "^7.2",
+        "php": "^7.4 | ^8.0",
```
Comment (Ocramius) [inline composer.json:8]: `|` is deprecated - should be `||`

**Review by Ocramius — APPROVED** [review]
LGTM :+1: 

Overall, it's a major improvement, and moves one of the last pending libraries off travis-ci, thanks @trejjam!

Checking builds, then merging :-)

File: Makefile (hunk 3 above)
Comment (trejjam) [reply Makefile:4]: It makes sense. I totally overlooked it. Thank you for the review!

**Pull request merged**